storage/src/nodestore.rs

```diff
@@ -225,6 +225,12 @@ impl<T: ReadInMemoryNode, S: ReadableStorage> NodeStore<T, S> {
         let node = Node::from_reader(area_stream)?;
         Ok(node.into())
     }
+
+    /// Put a [Node] into the cache at the given address.
+    fn cache_node(&self, addr: LinearAddress, node: Arc<Node>) -> Result<Arc<Node>, Error> {
+        self.storage.write_cached_nodes(once((&addr, &node)))?;
```

Comment:
Using `once` here is likely a performance hit. It has to set up an iterator and then iterate over them in `write_cached_nodes`. Since all it's really doing is calling `self.cache.insert` once, I'd suggest another method on the storage interface that just writes a single node.

The code was structured this way because it was more efficient to grab the lock once, but since that isn't the case any more and you can't write a block of data to the cache using the same lock, you could consider completely removing `write_cached_nodes` and just have a `write_cached_node` method everywhere.

My suspicion is that the locking that has to happen inside moka is complex and we win here because we only do the lock once.